app/src/main/java/org/oppia/android/app/profile/AddProfileActivityPresenter.kt

```diff
@@ -63,7 +63,7 @@ class AddProfileActivityPresenter @Inject constructor(
       lifecycleOwner = activity
       viewModel = profileViewModel
     }
-    if (!enableDownloadsSupport.value) {
+    if (enableDownloadsSupport.value) {
       binding.addProfileActivityAllowDownloadConstraintLayout.setOnClickListener {
         allowDownloadAccess = !allowDownloadAccess
```

Comment:
I still think the code on line 68 is ambiguous--

`allowDownloadAccess` is false by default. On ln68, we are saying ` allowDownloadAccess = !allowDownloadAccess` which means we are are negating the current value(false) and transfmorming it to true(ie. !false == true).
And if this is the intention of this expression, then can we use the boolean values `true/false` to convey the intended meaning?

Reply:
I am not very sure that I understand your question but since this listener is associated with a switch, and the user can toggle between on and off states many times it should not be trapped in either state forever.  That's why we use a negate logic other than assign it to a constant boolean value.